### PR TITLE
Leave Debug breadcrumbs for snapshotter functions called

### PR DESCRIPTION
**Pull request by dims**
When debugging issues with snapshotter, it was hard to figure out if anything was even called or not especially with `--log-level debug` should show some traffic in the logs

### CODE REVIEW

**Review by Kern-- — APPROVED** [review]

**Review by austinvazquez — APPROVED** [review]

File: snapshot/snapshot.go [hunk 1]
```diff
@@ -301,6 +305,7 @@ func (o *snapshotter) Prepare(ctx context.Context, key, parent string, opts ...s
 	//       or not, using the key `remoteSnapshotLogKey` defined in the above. This
 	//       log is used by tests in this project.
 	lCtx := log.WithLogger(ctx, log.G(ctx).WithField("key", key).WithField("parent", parent))
+	log.G(lCtx).WithField(remoteSnapshotLogKey, prepareSucceeded).Debug("preparing snapshot")
```
Comment (turan18) [inline snapshot/snapshot.go:308]: small change: Could we omit `WithField(remoteSnapshotLogKey, prepareSucceeded)` here. It should only be logged after we were able to perform a "remote" mount: https://github.com/awslabs/soci-snapshotter/pull/1036/files#diff-5c6bcd0ff5f76d3d6e311122b52e890cff76233ee29e88a138f074bcc74518e1R318

Comment (turan18) [inline snapshot/snapshot.go:308]: small change: Could we omit `WithField(remoteSnapshotLogKey, prepareSucceeded)` here. It should only be logged after we were able to perform a "remote" mount: https://github.com/awslabs/soci-snapshotter/pull/1036/files#diff-5c6bcd0ff5f76d3d6e311122b52e890cff76233ee29e88a138f074bcc74518e1R318

**Review by turan18 — APPROVED** [review]

File: snapshot/snapshot.go (hunk 1 above)
Comment (dims) [reply snapshot/snapshot.go:308]: Done!

**Review by turan18 — APPROVED** [review]
Thanks!

**Review by Kern-- — APPROVED** [review]

**Pull request merged**